scipy/stats/_continuous_distns.py

```diff
@@ -3247,6 +3247,114 @@ def _ppf(self, q, k, s):
 mielke = mielke_gen(a=0.0, name='mielke')
 
 
+class kappa4_gen(rv_continuous):
+    """Kappa 4 parameter distribution
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for kappa4 is::
+
+        kappa4.pdf(x, h, k) = (1.0 - k*x)**(1.0/k - 1.0)*
+                              (1.0 - h*(1.0 - k*x)**(1.0/k))**(1.0/h-1)
+
+    if ``h`` and ``k`` are not equal to 0.
+
+    kappa4 takes ``h`` and ``k`` as shape parameters, skew and
+    kurtosis respectively.
+
+    The kappa4 distribution return other distributions when certain
+    ``h`` and ``k`` values are used.
+
+    +------+-------------+--------------+---------------+
+    | h    | k=0.0       | k=1.0        | -2<=k<=2      |
+    +======+=============+==============+===============+
+    |  1.0 | Exponential | Uniform      | Generalized   |
+    |      |             |              | Pareto        |
+    +------+-------------+--------------+---------------+
+    |  0.0 | Gumbel      | Reverse      | Generalized   |
+    |      |             | Exponential  | Extreme Value |
+    |      |             | (note:not in |               |
+    |      |             | scipy)       |               |
+    +------+-------------+--------------+---------------+
+    | -1.0 | Logistic    |              | Generalized   |
+    |      |             |              | Logistic      |
+    +------+-------------+--------------+---------------+
+
+    %(after_notes)s
+
+    %(example)s
+
+    """
+    def _argcheck(self, h, k):
+        return (h >= -2) & (h <= 2) & (k >= -2) & (k <= 2)
```

Comment:
does this need restriction for bounded support, lower or upper limit ?  `.a` `.b`
should be parameter dependent,AFAICS